### ISSUE TITLE
HELP:  missingkeys and  gcc problem for xformers.

### ISSUE THREAD

**Issue by 3buffers** [cure-lab/MagicDrive#128]
Sorry to disturb your time, i have 2 problems

the first one is missing keys for train by :

accelerate launch --mixed_precision fp16 --gpu_ids all  --num_processes 2 tools/train.py +exp=224x400 runner=default

<img width="639" alt="miss" src="https://github.com/user-attachments/assets/7dd0e739-c460-423c-833a-5991e584ba21" />

is the sample problem with @about key missing when traing video #99?

CUDA=11.3
torch=1.10.2+cu113
python3.8
the second problem is  i cannot install xformer, and only can do test or train by without xformer

<img width="485" alt="torch" src="https://github.com/user-attachments/assets/d8896d6f-6389-4c92-a66b-e9a355b2444c" />

**Comment by flymin** [comment]
For the first problem, it is ok in most cases.

For the second, please problem more logs. The screenshot you captured did not show any useful information. If there were not more logs, please use `-vvv` option when you install.

**Comment by flymin** [comment]
Did you install our `xformers` or from the original source code? You cannot use the original version of `xformers`.

**Comment by 3buffers** [comment]
> Did you install our `xformers` or from the original source code? You cannot use the original version of `xformers`.

sorry not to attach log images, I have re-edited problem, sorry

**Comment by 3buffers** [comment]
> Did you install our `xformers` or from the original source code? You cannot use the original version of `xformers`.

I didn't install xformers and just use master version for magicdrive which includes xformers code

**Comment by 3buffers** [comment]
<img width="419" alt="prof" src="https://github.com/user-attachments/assets/6e4bb80b-5dc5-4cd3-8d37-cd349a6f3f79" />

i just print to show my torch vision at 114 line in setup.py, based on 113 line code, seems that I cannot load flash attention for xformers.

**Comment by flymin** [comment]
As you can see, this line only prints some information without triggering any error.

Please check the real error logs. Typically, they are located above the logs in your screenshot.

**Comment by 3buffers** [comment]
here is traceback  of image
<img width="592" alt="error" src="https://github.com/user-attachments/assets/6328433f-4beb-48db-9968-0fa78f687570" />

and here is full log.txt
[error_log.txt](https://github.com/user-attachments/files/18148354/error_log.txt)

thanks for your patience

**Comment by 3buffers** [comment]
Package                      Version        
---------------------------- -------------- 
absl-py                      2.1.0
accelerate                   0.20.3
addict                       2.4.0
aiofiles                     23.2.1
aliyun-python-sdk-core       2.16.0
aliyun-python-sdk-kms        2.16.5
altair                       5.4.1
annotated-types              0.7.0
antlr4-python3-runtime       4.9.3
anyio                        4.6.2
argon2-cffi                  23.1.0
argon2-cffi-bindings         21.2.0
arrow                        1.3.0
asttokens                    3.0.0
astunparse                   1.6.3
async-lru                    2.0.4
attrs                        24.2.0
babel                        2.16.0
backcall                     0.2.0
beautifulsoup4               4.12.3
bleach                       6.1.0
cachetools                   5.5.0
certifi                      2024.8.30
cffi                         1.17.1
charset-normalizer           3.4.0
click                        8.1.7
comm                         0.2.2
crcmod                       1.7
cycler                       0.12.1
debugpy                      1.8.11
decorator                    5.1.1
deepspeed                    0.16.1
defusedxml                   0.7.1
descartes                    1.1.0
diffusers                    0.17.1
einops                       0.3.2
exceptiongroup               1.2.2
executing                    2.1.0
fastapi                      0.115.6
fastjsonschema               2.21.1
ffmpy                        0.4.0
filelock                     3.16.1
fire                         0.7.0
flatbuffers                  24.3.25
fonttools                    4.55.3
fqdn                         1.5.1
fsspec                       2024.10.0
gast                         0.4.0
google-auth                  2.37.0
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
gradio                       4.16.0
gradio_client                0.8.1
grpcio                       1.68.1
h11                          0.14.0
h5py                         3.11.0
hjson                        3.1.0
httpcore                     1.0.7
httpx                        0.28.1
huggingface-hub              0.25.0
hydra-core                   1.3.0
idna                         3.10
importlib_metadata           8.5.0
importlib_resources          6.4.5
interval                     1.0.0
ipykernel                    6.29.5
ipython                      8.13.0
ipywidgets                   8.1.5
isoduration                  20.11.0
jedi                         0.19.2
Jinja2                       3.1.4
jmespath                     0.10.0
joblib                       1.4.2
json5                        0.10.0
jsonpointer                  3.0.0
jsonschema                   4.23.0
jsonschema-specifications    2023.12.1
jupyter                      1.1.1
jupyter_client               8.6.3
jupyter-console              6.6.3
jupyter_core                 5.7.2
jupyter-events               0.10.0
jupyter-lsp                  2.2.5
jupyter_server               2.14.2
jupyter_server_terminals     0.5.3
jupyterlab                   4.3.3
jupyterlab_pygments          0.3.0
jupyterlab_server            2.27.3
jupyterlab_widgets           3.0.13
keras                        2.10.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.7
libclang                     18.1.1
llvmlite                     0.36.0
loguru                       0.7.3
Markdown                     3.7
markdown-it-py               3.0.0
MarkupSafe                   2.1.5
matplotlib                   3.5.2
matplotlib-inline            0.1.7
mdurl                        0.1.2
mistune                      3.0.2
mmcv-full                    1.4.5
mmdet                        2.20.0
mmdet3d                      0.0.0          /home/ma-user/work/wufei/magicdrive/third_party/bevfusion
msgpack                      1.1.0
munkres                      1.1.4
narwhals                     1.18.3
nbclient                     0.10.1
nbconvert                    7.16.4
nbformat                     5.10.4
nest-asyncio                 1.6.0
ninja                        1.11.1.2
notebook                     7.3.1
notebook_shim                0.2.4
numba                        0.53.0
numpy                        1.21.6
nuscenes-devkit              1.1.10
nvidia-ml-py                 12.560.30
oauthlib                     3.2.2
omegaconf                    2.3.0
opencv-python                4.6.0.66
opencv-python-headless       4.10.0.84
opt_einsum                   3.4.0
orjson                       3.10.12
oss2                         2.19.1
overrides                    7.7.0
packaging                    24.2
pandas                       2.0.3
pandocfilters                1.5.1
parso                        0.8.4
pexpect                      4.9.0
pickle5                      0.0.11
pickleshare                  0.7.5
Pillow                       8.4.0
pip                          24.3.1
pkgutil_resolve_name         1.3.10
platformdirs                 4.3.6
prometheus_client            0.21.1
prompt_toolkit               3.0.48
protobuf                     3.19.6
psutil                       6.1.0
ptyprocess                   0.7.0
pure_eval                    0.2.3
py-cpuinfo                   9.0.0
pyasn1                       0.6.1
pyasn1_modules               0.4.1
pycocotools                  2.0.7
pycparser                    2.22
pycryptodome                 3.21.0
pydantic                     2.10.3
pydantic_core                2.27.1
pydub                        0.25.1
Pygments                     2.18.0
pyparsing                    3.1.4
pyquaternion                 0.9.9
python-dateutil              2.9.0.post0
python-json-logger           3.2.0
python-multipart             0.0.19
pytz                         2024.2
PyYAML                       6.0.2
pyzmq                        26.2.0
referencing                  0.35.1
regex                        2024.11.6
requests                     2.32.3
requests-oauthlib            2.0.0
rfc3339-validator            0.1.4
rfc3986-validator            0.1.1
rich                         13.9.4
rpds-py                      0.20.1
rsa                          4.9
ruff                         0.8.3
scikit-learn                 1.3.2
scipy                        1.10.1
seaborn                      0.12.2
semantic-version             2.10.0
Send2Trash                   1.8.3
setuptools                   59.5.0
Shapely                      1.8.5
shellingham                  1.5.4
six                          1.17.0
sniffio                      1.3.1
soupsieve                    2.6
stack-data                   0.6.3
starlette                    0.41.3
tensorboard                  2.10.0
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.10.0
tensorflow-estimator         2.10.0
tensorflow-io-gcs-filesystem 0.34.0
termcolor                    2.4.0
terminado                    0.18.1
terminaltables               3.1.10
threadpoolctl                3.5.0
tinycss2                     1.4.0
tokenizers                   0.13.3
tomli                        2.2.1
tomlkit                      0.12.0
torch                        1.10.2+cu113
torchvision                  0.11.3+cu113
tornado                      6.4.2
tqdm                         4.67.1
traitlets                    5.14.3
transformers                 4.27.4
typer                        0.15.1
types-python-dateutil        2.9.0.20241206
typing_extensions            4.12.2
tzdata                       2024.2
uri-template                 1.3.0
urllib3                      1.26.12
uvicorn                      0.33.0
wcwidth                      0.2.13
webcolors                    24.8.0
webencodings                 0.5.1
websocket-client             1.8.0
websockets                   11.0.3
Werkzeug                     3.0.6
wheel                        0.45.1
widgetsnbextension           4.0.13
wrapt                        1.17.0
yapf                         0.43.0
zipp                         3.20.2

**Comment by 3buffers** [comment]
> > gcc: internal compiler error: Segmentation fault (program cc1plus)
> 
> I think this error is the key. However, I do not have concrete solution. You may want to try upgrade/re-install gcc and g++. Otherwise, please use Google.

thanks for your reply, may i know your gcc/g++ version, and I also will use google , in final thanks for your time to reply my issue and i will close this issue

**Comment by flymin** [comment]
You can find some information
[here](https://github.com/cure-lab/MagicDrive/blob/main/doc/FAQ.md#q2-how-to-setup-on-ampere-gpus-eg-a100-a800). Typically, if the version is incompatible, you can find clear error information rather than segmentation fault. Such error may indicate the binary file itself is corrupted or incompatible with the OS.

**Comment by 3buffers** [comment]
> You can find some information [here](https://github.com/cure-lab/MagicDrive/blob/main/doc/FAQ.md#q2-how-to-setup-on-ampere-gpus-eg-a100-a800). Typically, if the version is incompatible, you can find clear error information rather than segmentation fault. Such error may indicate the binary file itself is corrupted or incompatible with the OS.

thanks, i runed successfully by upgrade gcc/g++ to 9 version, thanks.

**Comment by 3buffers** [comment]
successfully by upgrade gcc/g++ to 9 version  for xformer compiling
